### PR TITLE
fix: [MEW-1912] validate take-profit / stop-loss before saving

### PR DESCRIPTION
**Pull request by renerpdev**
## Summary

The Auto Close (Take Profit / Stop Loss) dialog only validated price precision, so a user could set a take-profit of `0` (e.g. the HOOD `+100%` pill on a short computes to `mark * (1 - 1) = 0`), or a TP/SL on the wrong side of the mark price, and still hit **Save**. The backend then rejected the order only at confirm time (`take profit trigger price '0' is invalid – must be positive`).

## Changes

- **`src/modules/perps/utils/tpSlValidation.ts`** (new): pure, side-aware validators returning an error message (or `''` when valid):
  - **TP** — must be positive, `< $10,000,000`, and above the mark for a **long** / below the mark for a **short**.
  - **SL** — must be positive, and below the mark for a **long** / above the mark for a **short**.
  - Precision errors take priority (message preserved).
- **`src/modules/perps/composables/usePerpsTradeForm.ts`**: adds `isTpSlLong` and `takeProfitErrorMessage` / `stopLossErrorMessage` (replacing the precision-only booleans); `submitDisabled` now gates on these so an invalid TP/SL can't be submitted at all. The `+/-%` pill setters reuse `isTpSlLong`.
- **`src/modules/perps/components/PerpsTakeProfitStopLossDialog.vue`**: displays the real error message per field and disables **Save** while either TP or SL is invalid.

### Note on long vs short direction

The ticket lists the long-side messages (TP above mark, SL below mark). Since the reported case is a **short** position, the direction is implemented **side-aware** (short inverts to: TP below mark, SL above mark) — a "TP above mark" on a short would lock in a loss, so the literal long-only messages would be incorrect there.

## Testing

- `pnpm vitest run tests/unit/specs/modules/perps/utils/tpSlValidation.spec.ts` → 16/16 pass (long/short, zero, wrong-side, `$10M`, precision)
- `pnpm vue-tsc --noEmit -p tsconfig.app.json` → clean
- `pnpm eslint` on changed files → clean
- Manual smoke test on long and short: TP=0 / wrong-side / `≥ $10M` and SL wrong-side / `≤ 0` each show the error and disable Save; valid values save and confirm without a backend rejection.

_(Pre-existing, unrelated: `usePerpsAuth.spec.ts` fails to load locally due to a missing `@ledgerhq/hw-app-eth` module in the pnpm store — fails identically on clean HEAD.)_

## Summary by CodeRabbit

* **New Features**
  * Added client-side TP/SL trigger validation for Auto Close in perps trading, producing precise, user-facing error messages.
* **Bug Fixes**
  * Updated the TP/SL dialog to display validation messages directly and disable saving when either TP or SL is invalid.
  * Improved long/short TP/SL placement rules and prevented invalid TP/SL drafts from affecting the main submission state.
* **Tests**
  * Added unit tests covering TP/SL validation, including precision handling and boundary cases.

### CODE REVIEW

File: src/modules/perps/composables/usePerpsTradeForm.ts [hunk 1]
```diff
@@ -591,21 +592,34 @@ export function usePerpsTradeForm() {
...
-  const takeProfitPrecisionError = computed(() => {
-    if (tempTakeProfitPrice.value == null) return false
-    return hasInvalidPrecision(
-      String(tempTakeProfitPrice.value),
+  // Take-profit / stop-loss direction: long → TP above mark, SL below; short
+  // inverts both. Mirrors the isLong used by the +/-% pill helpers below.
+  const isTpSlLong = computed(
+    () =>
+      activePosition.value?.direction === 'long' || orderSide.value === 'buy',
+  )
```
[the remark this answers is not in the document]
Comment (renerpdev) [reply src/modules/perps/composables/usePerpsTradeForm.ts:601]: Done in 1e16a5b899 — `isTpSlLong` now derives direction from the active position first (`activePosition.value ? activePosition.value.direction === 'long' : orderSide.value === 'buy'`), so it no longer depends on the external watcher keeping `orderSide` in sync. The truthy check also guards `.direction`, so there's no null access.

File: src/modules/perps/composables/usePerpsTradeForm.ts [hunk 2]
```diff
@@ -591,21 +592,34 @@ export function usePerpsTradeForm() {
...
+  const stopLossErrorMessage = computed(() =>
+    stopLossError(
+      tempStopLossPrice.value,
+      currentPrice.value,
+      isTpSlLong.value,
       quoteDecimals.value,
-    )
-  })
+    ),
+  )
```
[the remark this answers is not in the document]
Comment (renerpdev) [reply src/modules/perps/composables/usePerpsTradeForm.ts:623]: Done in e0f7a7a5d6 — when the Auto Close modal closes without saving, the temp TP/SL refs (and pill highlights) are reset back to the committed values in the existing `showAutoCloseModal` watcher. An invalid draft left on cancel no longer keeps `submitDisabled` (which keys off the temp refs) stuck on the main Long/Short button.

**Review by olgakup — APPROVED** [review]
